edxval/tests/test_api.py

```diff
@@ -1403,6 +1411,229 @@ def test_external_no_video_transcript(self):
+
+        # Verify that 2 new transcripts are created.
+        self.assertEqual(VideoTranscript.objects.count(), 2)
+
+        # Newly created video record.
+        edx_video_id = videos[1].edx_video_id
+
+        # Verify trancript records are created with correct data.
+        expected_transcripts =  [
+            dict(constants.VIDEO_TRANSCRIPT_CUSTOM_SRT, video_id=edx_video_id, language_code='en'),
+            dict(constants.VIDEO_TRANSCRIPT_CUSTOM_SRT, video_id=edx_video_id, language_code='es')
+        ]
+
+        self.assert_transcripts(
+            edx_video_id,
+            expected_transcripts
+        )
+
+    def test_multiple_contentstore_transcripts_for_language(self):
+        """
+        Verify that transcript import for external video working as expected when multiple transcripts present against
+        a language e.g. external english transcript is imported through sub and transcripts field.
+        """
+        # First create external transcripts.
+        sub_transcript_file_name = 'external-transcript-sub.srt'
+        utils.create_file_in_fs(
+            constants.VIDEO_TRANSCRIPT_CUSTOM_SRT['file_data'],
+            sub_transcript_file_name,
+            self.file_system,
+            constants.EXPORT_IMPORT_STATIC_DIR
+        )
+
+        ext_transcript_file_name = 'external-transcript-ext.sjson'
+        utils.create_file_in_fs(
+            constants.VIDEO_TRANSCRIPT_CUSTOM_SJSON['file_data'],
+            ext_transcript_file_name,
+            self.file_system,
+            constants.EXPORT_IMPORT_STATIC_DIR
+        )
+
+        # Verify that one video is present before import.
+        video_initial_count = 1
+        self.assertEqual(Video.objects.count(), video_initial_count)
+
+        # Verify that no transript was present before import.
+        self.assertEqual(VideoTranscript.objects.count(), 0)
+
+        # Import xml with empty edx video id.
+        api.import_from_xml(
+            etree.fromstring('<video_asset/>'),
+            '',
+            self.file_system,
+            constants.EXPORT_IMPORT_STATIC_DIR,
+            {
+                'en': [sub_transcript_file_name, ext_transcript_file_name]
+            }
+        )
+
+        # Verify that new video is created.
+        videos = Video.objects.all()
+        self.assertEqual(len(videos), video_initial_count + 1)
+
+        # Verify that only 1 new transcripts is created.
+        self.assertEqual(VideoTranscript.objects.count(), 1)
+
+        # Newly created video record.
+        edx_video_id = videos[1].edx_video_id
+
+        # Verify trancript record is created with correct data i.e sub field transcript.
+        expected_transcripts =  [
+            dict(constants.VIDEO_TRANSCRIPT_CUSTOM_SRT, video_id=edx_video_id, language_code='en')
+        ]
+
+        self.assert_transcripts(
+            edx_video_id,
+            expected_transcripts
+        )
+
+    def test_external_internal_transcripts_priority(self):
```

Comment:
better: `test_external_internal_transcripts_conflict` and also, `test_external_internal_transcripts_different_languages` is missing.